data/transactions/transaction_test.go

```diff
@@ -447,6 +447,37 @@ func TestWellFormedErrors(t *testing.T) {
 			proto:         futureProto,
 			expectedError: fmt.Errorf("tx has too many references, max is 8"),
 		},
+		{
+			tx: Transaction{
+				Type:   protocol.ApplicationCallTx,
+				Header: okHeader,
+				ApplicationCallTxnFields: ApplicationCallTxnFields{
+					ApplicationID:     1,
+					ApprovalProgram:   []byte(strings.Repeat("X", 1025)),
+					ClearStateProgram: []byte(strings.Repeat("X", 1025)),
+					ExtraProgramPages: 0,
+					OnCompletion:      UpdateApplicationOC,
+				},
+			},
+			spec:          specialAddr,
+			proto:         curProto,
```

Comment:
this will fail after protocol switch, please use a concrete version

Reply:
this is because it's an app update. the check for program lens for an app update has moved to https://github.com/shiqizng/go-algorand/blob/fix/exPages/ledger/apply/application.go#L204-L210